stress: failed test in cockroach/sql/sql.test: TestRaceWithBackfill

Binary: cockroach/static-tests.tar.gz sha: https://github.com/cockroachdb/cockroach/commits/5230e77e7c2d942255c51876f1e0edbb960fd9bf

Stress build found a failed test:

```
=== RUN   TestRaceWithBackfill
I160527 07:07:56.558053 storage/engine/rocksdb.go:137  opening in memory rocksdb instance
W160527 07:07:56.560815 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160527 07:07:56.561666 server/node.go:401  store store=0:0 ([]=) not bootstrapped
W160527 07:07:56.562717 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160527 07:07:56.563057 storage/replica_command.go:1470  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.562847579s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.562998733 +0000 UTC]
I160527 07:07:56.563368 server/node.go:327  **** cluster {c9c9f25e-cd6e-4885-993d-0a6be8707d2e} has been created
I160527 07:07:56.563519 server/node.go:328  **** add additional nodes by specifying --join=127.0.0.1:38430
W160527 07:07:56.563544 gossip/gossip.go:897  not connected to cluster; use --join to specify a connected node
I160527 07:07:56.564006 server/node.go:414  initialized store store=1:1 ([]=): {Capacity:10365558784 Available:7964147712 RangeCount:0}
I160527 07:07:56.564025 server/node.go:302  node ID 1 initialized
I160527 07:07:56.564084 storage/stores.go:287  read 0 node addresses from persistent storage
I160527 07:07:56.564161 server/node.go:535  connecting to gossip network to verify cluster ID...
I160527 07:07:56.565449 server/node.go:556  node connected via gossip and verified as part of cluster {"c9c9f25e-cd6e-4885-993d-0a6be8707d2e"}
I160527 07:07:56.565486 server/node.go:355  [node=1] Started node with [[]=] engine(s) and attributes []
I160527 07:07:56.565519 server/server.go:418  starting https server at 127.0.0.1:36629
I160527 07:07:56.565537 server/server.go:419  starting grpc/postgres server at 127.0.0.1:38430
I160527 07:07:56.565908 storage/split_queue.go:100  splitting range=1 [/Min-/Max) at keys [/Table/11/0 /Table/12/0 /Table/13/0 /Table/14/0]
I160527 07:07:56.609419 storage/replica_command.go:2011  initiating a split of range=1 [/Min-/Max) at key /Table/11
I160527 07:07:56.609640 server/updates.go:155  No previous updates check time.
I160527 07:07:56.624245 storage/replica_command.go:1470  range 2: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.674019251s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.624179224 +0000 UTC]
I160527 07:07:56.625936 storage/replica_command.go:2011  initiating a split of range=2 [/Table/11-/Max) at key /Table/12
I160527 07:07:56.631701 storage/replica_command.go:1470  range 3: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.6814901s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.631655199 +0000 UTC]
I160527 07:07:56.631815 storage/replica_command.go:2011  initiating a split of range=3 [/Table/12-/Max) at key /Table/13
W160527 07:07:56.637428 storage/intent_resolver.go:299  [node=1,store=1,range=1] failed to push during intent resolution: failed to push "storage/replica_command.go:2013 (*Replica).AdminSplit" id=81ab311e key=/Local/Range/"\x95"/RangeDescriptor rw=true pri=0.00534984 iso=SERIALIZABLE stat=PENDING epo=0 ts=1464332876.631895906,0 orig=1464332876.631895906,0 max=1464332876.631895906,0 wto=false rop=false
I160527 07:07:56.653882 storage/replica_command.go:1470  range 4: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.690029176s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.653843679 +0000 UTC]
I160527 07:07:56.654106 storage/replica_command.go:2011  initiating a split of range=4 [/Table/13-/Max) at key /Table/14
I160527 07:07:56.740704 storage/replica_command.go:1470  range 5: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.790488529s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.740660692 +0000 UTC]
I160527 07:07:56.740810 storage/split_queue.go:100  splitting range=5 [/Table/14-/Max) at keys [/Table/50/0]
I160527 07:07:56.741086 storage/replica_command.go:2011  initiating a split of range=5 [/Table/14-/Max) at key /Table/50
I160527 07:07:56.760836 storage/replica_command.go:1470  range 6: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.810571649s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.760789122 +0000 UTC]
I160527 07:07:56.765971 storage/split_queue.go:100  splitting range=6 [/Table/50-/Max) at keys [/Table/51/0]
I160527 07:07:56.776507 storage/replica_command.go:2011  initiating a split of range=6 [/Table/50-/Max) at key /Table/51
I160527 07:07:56.799723 storage/replica_command.go:1470  range 7: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC 406759h7m57.849501181s following replica {0 0 0} 1970-01-01 00:00:00 +0000 UTC 0 [physicalTime=2016-05-27 07:07:56.799678835 +0000 UTC]
I160527 07:07:57.565319 gossip/gossip.go:923  starting client to 127.0.0.1:38430
I160527 07:07:57.595887 gossip/client.go:86  closing client to node 1 (127.0.0.1:38430): gossip/client.go:180: stopping outgoing client to node 1 (127.0.0.1:38430); loopback connection
I160527 07:07:57.782408 sql/lease.go:247  publish (count leases): descID=51 name=test version=4 count=1
I160527 07:07:57.833836 sql/lease.go:247  publish (count leases): descID=51 name=test version=5 count=1
I160527 07:07:58.239815 sql/lease.go:247  publish (count leases): descID=51 name=test version=6 count=1
I160527 07:07:58.359923 sql/lease.go:247  publish (count leases): descID=51 name=test version=7 count=1
I160527 07:07:58.429092 sql/lease.go:247  publish (count leases): descID=51 name=test version=8 count=1
E160527 07:07:58.545203 storage/replica.go:2229  [node=1,store=1,range=1] could not load SystemConfig span: must retry later due to intent on SystemConfigSpan
I160527 07:07:58.845034 sql/lease.go:247  publish (count leases): descID=51 name=test version=9 count=1
I160527 07:07:58.965092 sql/lease.go:247  publish (count leases): descID=51 name=test version=10 count=1
--- FAIL: TestRaceWithBackfill (2.85s)
	schema_changer_test.go:469: schema change ALTER TABLE t.test ADD COLUMN x DECIMAL DEFAULT (DECIMAL '1.4') took 487.755784ms
	schema_changer_test.go:469: schema change ALTER TABLE t.test DROP pi took 517.242486ms
	schema_changer_test.go:469: schema change CREATE UNIQUE INDEX foo ON t.test (v) took 210.594634ms
	schema_changer_test.go:535: expected 20055 key value pairs, but got 16074
```

Run Details:

```
0 runs so far, 0 failures, over 5s
0 runs so far, 0 failures, over 10s
0 runs so far, 0 failures, over 15s
0 runs so far, 0 failures, over 20s
0 runs so far, 0 failures, over 25s
0 runs so far, 0 failures, over 30s
0 runs so far, 0 failures, over 35s
0 runs so far, 0 failures, over 40s
12 runs so far, 0 failures, over 45s
16 runs so far, 0 failures, over 50s
16 runs so far, 0 failures, over 55s
16 runs so far, 0 failures, over 1m0s
16 runs so far, 0 failures, over 1m5s
16 runs so far, 0 failures, over 1m10s
16 runs so far, 0 failures, over 1m15s
16 runs so far, 0 failures, over 1m20s
16 runs so far, 0 failures, over 1m25s
27 runs so far, 0 failures, over 1m30s
32 runs so far, 0 failures, over 1m35s
32 runs so far, 0 failures, over 1m40s
32 runs so far, 0 failures, over 1m45s
32 runs so far, 0 failures, over 1m50s
32 runs so far, 0 failures, over 1m55s
32 runs so far, 0 failures, over 2m0s
32 runs so far, 0 failures, over 2m5s
33 runs so far, 0 failures, over 2m10s
46 runs so far, 0 failures, over 2m15s
48 runs so far, 0 failures, over 2m20s
48 runs so far, 0 failures, over 2m25s
48 runs so far, 0 failures, over 2m30s
48 runs so far, 0 failures, over 2m35s
48 runs so far, 0 failures, over 2m40s
48 runs so far, 0 failures, over 2m45s
48 runs so far, 0 failures, over 2m50s
50 runs so far, 0 failures, over 2m55s
61 runs so far, 0 failures, over 3m0s
64 runs so far, 0 failures, over 3m5s
64 runs so far, 0 failures, over 3m10s
64 runs so far, 0 failures, over 3m15s
64 runs so far, 0 failures, over 3m20s
64 runs so far, 0 failures, over 3m25s
64 runs so far, 0 failures, over 3m30s
65 runs completed, 1 failures, over 3m34s
FAIL
```
Please assign, take a look and update the issue accordingly.

--- a/sql/scan.go
+++ b/sql/scan.go
@@ -181,7 +181,7 @@ func (n *scanNode) Next() (bool, error) {
 	tracing.AnnotateTrace()
 	if !n.scanInitialized {
 		if err := n.initScan(); err != nil {
-			return false, nil
+			return false, err
 		}
 	}
 

